### DISCUSSION TITLE
sudo: The "no new privileges" flag is set, which prevents sudo from running as root.

### DISCUSSION THREAD

**Discussion by conor-nsurely** [gis-ops/docker-valhalla#87] [Q&A]
Hello,

I am using `gisops/valhalla:latest` to build a container to deploy to GCP Cloudrun. It fails due to this permissions error.
'sudo: The "no new privileges" flag is set, which prevents sudo from running as root.'

The container works locally and on  Kubernetes so perhaps this flag is set on the GCP cloudrun side.

My Dockerfile is simply
```
FROM gisops/valhalla:latest
COPY ./custom_files /custom_files
```

Attempting to run the container using the root user( As a test to bypass the flag) results in this error.

'ERROR: If you run with custom UID or GID you have to create the mapped directory to the container's /custom_files manually before starting the image'

Does this mean to simply add `RUN mkdir custom_files` to the Dockerfile and copy the files in there or?

e.g
```
FROM gisops/valhalla:latest
USER root
RUN mkdir custom_files
COPY ./custom_files /custom_files
```

Any guidance would be appreciated.

Thanks!

**Comment by nilsnolde** [comment]
Sorry, no idea about vendor-specific quirks. I'll convert this to a discussion as it has little to do with the actual project.

**Reply by conor-nsurely** [reply]
No worries @nilsnolde . 

More specifically could you clarify what is to be done in regards to 

`
ERROR: If you run with custom UID or GID you have to create the mapped directory to the container's /custom_files manually before starting the image`


```

FROM gisops/valhalla:latest
USER root
RUN mkdir custom_files
COPY ./custom_files /custom_files
```
This doesn't work. I may be misunderstanding the above errors message.


That would be enough resolve my issue temporarily.

**Comment by jcardus** [comment]
I had this problem, it went away when I changed the execution enviroment to second generation:

![Screenshot 2023-10-07 at 17 30 41](https://github.com/gis-ops/docker-valhalla/assets/31902485/822d1c0e-832d-4971-a9f2-82ffe5b2ee67)
